bftengine/src/bftengine/DbCheckpointManager.cpp

```diff
@@ -181,10 +181,30 @@ void DbCheckpointManager::initializeDbCheckpointManager(std::shared_ptr<concord:
     cleanUpFuture_ = std::async(std::launch::async, [this]() { cleanUp(); });
   }
 }
+
 std::optional<DbCheckpointMetadata::DbCheckPointDescriptor> DbCheckpointManager::getLastCreatedDbCheckpointMetadata() {
   std::scoped_lock lock(lockLastDbCheckpointDesc_);
   return lastCreatedCheckpointMetadata_;
 }
+
+DbCheckpointManager::CheckpointState DbCheckpointManager::getCheckpointState(CheckpointId id) const {
+  {
+    auto lock = std::scoped_lock{lockDbCheckPtFuture_};
```

Comment:
Should we check for if "(dbCheckptMetadata_.dbCheckPoints_.count(id))" first. Because "(dbCreateCheckPtFuture_.count(id))" will be equal to 1 in both cases pending and created

Reply:
Yes, you are right. I was assuming we remove the future when we complete creating the checkpoint, but we actually remove it when we remove the oldest checkpoint. Let me fix it.

Reply:
I am wondering whether I should just change the order of locks or whether I should keep `lock_` while locking `lockDbCheckPtFuture_? Based on the usage, it seems when we change the futures map, we always hold both locks anyway.